src/categorical_algebra/CSetDataStructures.jl

```diff
@@ -857,4 +858,37 @@ function init_acset(T::Type{<:ACSet{CD,AD,Ts}},body) where {CD <: CatDesc, AD <:
   code
 end
 
+""" Dataframes backed by ACSets
+
+Useful as a "view" onto an ACSet, and also for computing limits
+"""
+
+""" P is a symbol, the object in the schema that this ASet focuses onto
+"""
+struct ACSetView{A <: ACSet, P, Attrs <: NamedTuple} <: AbstractArray{Attrs,1}
+  backing :: A
+  parts :: Vector{Int}
```

Comment:
I figure that we should allow `AbstractVector{Int}` so that we can use ranges (`1:n`) without allocating the corresponding array.

Reply:
Ah, this is definitely a good idea. I was updating this to be a union of `Vector{Int}` and `Colon`, but it's much better to use AbstractVector; that simplifies things a lot.